workflow/common/util_test.go

```diff
@@ -290,3 +292,111 @@ func TestOverridableDefaultInputArts(t *testing.T) {
+		}, true, nil
+	}
+
+	tmpl.Inputs.Parameters = []wfv1.Parameter{{Name: paramName, Value: wfv1.AnyStringPtr("abc")}}
+
+	valueArgs := wfv1.Inputs{Parameters: []wfv1.Parameter{{Name: paramName, Value: wfv1.AnyStringPtr("override")}}}
+	valueFromArgs := wfv1.Inputs{Parameters: []wfv1.Parameter{{Name: paramName, ValueFrom: &wfv1.ValueFrom{ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+		LocalObjectReference: corev1.LocalObjectReference{
+			Name: overrideConfigMapName,
+		},
+		Key: overrideConfigMapKey,
+	}}}}}
+
+	globalParams := make(map[string]string)
+	localParams := make(map[string]string)
+
+	newTmpl, err := ProcessArgs(&tmpl, &valueArgs, globalParams, localParams, false, "", configMapStore)
+	assert.Nil(t, err)
+	assert.NotNil(t, newTmpl)
+	assert.Equal(t, newTmpl.Inputs.Parameters[0].Value.String(), valueArgs.Parameters[0].Value.String())
+
+	newTmpl, err = ProcessArgs(&tmpl, &valueFromArgs, globalParams, localParams, false, "", configMapStore)
+	assert.Nil(t, err)
+	assert.NotNil(t, newTmpl)
+	assert.Equal(t, newTmpl.Inputs.Parameters[0].Value.String(), overrideConfigMapValue)
+}
+
+func TestOverridableTemplateInputParamsValueFrom(t *testing.T) {
+	tmpl := wfv1.Template{}
+	tmpl.Name = "artifact-printing"
+
+	paramName := "value-from-param"
+
+	configMapName := "config-map-name"
+	configMapKey := "config-map-key"
+	configMapValue := "config-map-value"
+
+	overrideConfigMapName := "override-config-map-name"
+	overrideConfigMapKey := "override-config-map-key"
+	overrideConfigMapValue := "override-config-map-value"
+
+	configMapStore := mockConfigMapStore{}
+	configMapStore.getByKey = func(key string) (interface{}, bool, error) {
+		return &apiv1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{LabelKeyConfigMapType: LabelValueTypeConfigMapParameter}},
+			Data: map[string]string{configMapKey: configMapValue, overrideConfigMapKey: overrideConfigMapValue},
+		}, true, nil
+	}
+
+	tmpl.Inputs.Parameters = []wfv1.Parameter{{Name: paramName, ValueFrom: &wfv1.ValueFrom{
+		ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: configMapName,
+			},
+			Key: configMapKey,
+		},
+	}}}
+
+	valueArgs := wfv1.Inputs{Parameters: []wfv1.Parameter{{Name: paramName, Value: wfv1.AnyStringPtr("override")}}}
+	valueFromArgs := wfv1.Inputs{Parameters: []wfv1.Parameter{{
+		Name: paramName,
+		ValueFrom: &wfv1.ValueFrom{
+			ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: overrideConfigMapName,
+				},
+				Key: overrideConfigMapKey,
+			},
+		},
+	}}}
+
+	globalParams := map[string]string{paramName: "overrideValue"}
+	localParams := make(map[string]string)
+
+	newTmpl, err := ProcessArgs(&tmpl, &valueArgs, globalParams, localParams, false, "", configMapStore)
+	assert.Nil(t, err)
+	assert.NotNil(t, newTmpl)
+	assert.Equal(t, newTmpl.Inputs.Parameters[0].Value.String(), valueArgs.Parameters[0].Value.String())
```

Comment:
The original code will fail at this assertion as explained in the [issue](https://github.com/argoproj/argo-workflows/issues/10182)